pkg/store/file_log_dcs.go

```diff
@@ -0,0 +1,127 @@
+package store
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/pkg/errors"
+	"storj.io/uplink"
+)
+
+const uploadTimeout = 15 * time.Second
+
+var (
+	_ Log          = (*fileLogDCS)(nil)
+	_ TrashSegment = (fileTrashSegmentDCS{})
+)
+
+// fileLogDCS wraps fileLog and intercepts Purgeable to wrap fileTrashSegment
+// with fileTrashSegmentDCS.
+type fileLogDCS struct {
```

Comment:
This is super cool! Amazing it fits into the existing code so easily like this.

Reply:
Good call; thanks, Sean!

Reply:
Right? It's impressive how well structured OK Log is.